prow/cmd/sinker/main.go

```diff
@@ -124,48 +124,35 @@ func (c *controller) clean() {
 		return
 	}
 
-	// Only delete pod if its prowjob is marked as finished
-	isFinished := make(map[string]bool)
-
-	maxProwJobAge := c.configAgent.Config().Sinker.MaxProwJobAge
-	for _, prowJob := range prowJobs {
-		// Handle periodics separately.
-		if prowJob.Spec.Type == kube.PeriodicJob {
-			continue
-		}
-		if !prowJob.Complete() {
-			continue
-		}
-		isFinished[prowJob.ObjectMeta.Name] = true
-		if time.Since(prowJob.Status.StartTime.Time) <= maxProwJobAge {
-			continue
-		}
-		if err := c.kc.DeleteProwJob(prowJob.ObjectMeta.Name); err == nil {
-			c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Deleted prowjob.")
-		} else {
-			c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).WithError(err).Error("Error deleting prowjob.")
-		}
-	}
-
 	// Keep track of what periodic jobs are in the config so we will
 	// not clean up their last prowjob.
+	// This allows horologium to determine time since the last run for periodics.
 	isActivePeriodic := make(map[string]bool)
 	for _, p := range c.configAgent.Config().Periodics {
 		isActivePeriodic[p.Name] = true
 	}
-
-	// Get the jobs that we need to retain so horologium can continue working
-	// as intended.
 	latestPeriodics := pjutil.GetLatestProwJobs(prowJobs, kube.PeriodicJob)
-	for _, prowJob := range prowJobs {
-		if prowJob.Spec.Type != kube.PeriodicJob {
-			continue
-		}
 
-		latestPJ := latestPeriodics[prowJob.Spec.Job]
-		if isActivePeriodic[prowJob.Spec.Job] && prowJob.ObjectMeta.Name == latestPJ.ObjectMeta.Name {
-			// Ignore deleting this one.
-			continue
+	// Only delete pod if either
```

Comment:
Didn't we discuss allowing prowjobs to specify that running pods should be deleted as soon as the prow job is aborted and/or no longer running (such as when a new commit is pushed)? It seems like this doesn't handle that scenario yet.

(and we also need a way for prowjobs to specify that pods should always be given time to complete).